### DISCUSSION TITLE
misunderstanding of the rack option

### DISCUSSION THREAD

**Discussion by f4rx** [minio/minio#18707] [Q&A] [answered]
Unfortunately, I can't find a description of the rack option in the documentation.

I'm simulating the situation - there are only 6 servers, and 3 racks. Thus, there are two servers in the rack. Let's say there are two disks in each server.

My goal is for the system to survive a single rack failure.

https://min.io/product/erasure-code-calculator offers the following options:
Command line options:
```bash
minio server https://rack{1...3}.node{1...2}.example.net/mnt/drive{1...2}
```

But I didn't find what rack is in the documentation. Also, the web interface and the CLI do not offer a view of which rack the servers are located in.
Will this guarantee availability in case of failure of two servers in the same rack?
Or I can omit the rack option, and set the option `MINIO_STORAGE_CLASS_STANDARD=EC:6` (or EC:4 ?) - will be enough?

**Comment by harshavardhana** [comment, marked as answer]
A rack is a server rack that holds your servers. So if you have 3 racks with 2 nodes each that's what your above command looks like. 


> Or I can omit the rack option, and set the option `MINIO_STORAGE_CLASS_STANDARD=EC:6` (or EC:4 ?) - will be enough?

This option is for parity not number of disks in an stripe set.

**Reply by ravindk89** [reply]
The deeper dive here is a little more nuanced, and it's on our todo list to try to document more thoroughly @f4rx .

Break up the command:

```
minio server https://rack{1...3}.node{1...2}.example.net/mnt/drive{1...2}
                                    | rack         |      node   |          TLD   |      drives           |
```

With our logic, when we calculate erasure set drives and distribute drives, that distribution will respect the above topology - 3 racks, with 2 nodes per rack, and 2 drives per node.

This conceptually results in in a distribution of erasure sets across racks *and* nodes, in such a way to optimize resiliency in the event of rack failure.

For these to work _in practice_ though we'd probably want to see more nodes and drives in use. The fewer nodes and drives you have, the less effective any mitigations of rack distribution.

There isn't to my knowledge a way to visualize this topology at this time, though it would certainly be interesting.
